File: Python Fundamentals/Mid Exam problems/bonus_scoring_system.py
from math import ceil

students = int(input())
lectures = int(input())
bonus = int(input())

max_points = 0
max_student_lecture = 0

for student in range(students):

    attendances = int(input())
    total_bonus = (attendances / lectures) * (5 + bonus)

    if total_bonus >= max_points:
        max_points = total_bonus
        max_student_lecture = attendances

print(f"Max Bonus: {ceil(max_points)}.")
print(f"The student has attended {max_student_lecture} lectures.")

File: JavaScript Fundamentals/Mid Exams/Mid Exam 5/bonusScoringSystem.js
function solve(arr) {
    let lectures = Number(arr.shift());
    let bonus = Number(arr.shift());
    let points = 0;
    let maxPoints = 0;
    let studentAttendances = 0;

    for (let attendance of arr) {
        points = Number(attendance) / lectures * (5 + bonus);
        if (points > maxPoints) {
            maxPoints = points;
            studentAttendances = attendance;
        }
    }

    console.log(`Max Bonus: ${Math.ceil(maxPoints)}.`);
    console.log(`The student has attended ${studentAttendances} lectures.`);
}
